idaes/core/util/model_diagnostics.py

```diff
@@ -17,29 +17,1023 @@
+        xjr = extreme_jacobian_rows(
+            m=self._model,
+            scaled=False,
+            large=self.config.jacobian_large_value_caution,
+            small=self.config.jacobian_small_value_caution,
+        )
+        xjr.sort(key=lambda i: abs(log(i[0])), reverse=True)
+
+        _write_report_section(
+            stream=stream,
+            lines_list=[f"{i[1].name}: {i[0]:.3E}" for i in xjr],
+            title="The following constraints(s) are associated with extreme Jacobian values:",
+            header="=",
+            footer="=",
+        )
+
+    def display_extreme_jacobian_entries(self, stream=stdout):
+        """
+        Prints variables and constraints associated with entries in the Jacobian with extreme
+        values. This can be indicative of poor scaling, especially for isolated terms (e.g.
+        variables which appear only in one term of a single constraint).
+
+        Tolerances can be set via the DiagnosticsToolbox config.
+
+        Args:
+            stream: an I/O object to write the output to (default = stdout)
+
+        Returns:
+            None
+
+        """
+        xje = extreme_jacobian_entries(
+            m=self._model,
+            scaled=False,
+            large=self.config.jacobian_large_value_caution,
+            small=self.config.jacobian_small_value_caution,
+            zero=0,
+        )
+        xje.sort(key=lambda i: abs(log(i[0])), reverse=True)
+
+        _write_report_section(
+            stream=stream,
+            lines_list=[f"{i[1].name}, {i[2].name}: {i[0]:.3E}" for i in xje],
+            title="The following constraints(s) and variable(s) are associated with extreme Jacobian\nvalues:",
+            header="=",
+            footer="=",
+        )
+
+    # TODO: Block triangularization analysis
+    # Number and size of blocks, polynomial degree of 1x1 blocks, simple pivot check of moderate sized sub-blocks?
+
+    def _collect_structural_warnings(self):
+        """
+        Runs checks for structural warnings and returns two lists.
+
+        Returns:
+            warnings - list of warning messages from structural analysis
+            next_steps - list of suggested next steps to further investigate warnings
+
+        """
+        uc = identify_inconsistent_units(self._model)
+        uc_var, uc_con, oc_var, oc_con = self.get_dulmage_mendelsohn_partition()
+
+        # Collect warnings
+        warnings = []
+        next_steps = []
+        dof = degrees_of_freedom(self._model)
+        if dof != 0:
+            dstring = "Degrees"
+            if abs(dof) == 1:
+                dstring = "Degree"
+            warnings.append(f"WARNING: {dof} {dstring} of Freedom")
+        if len(uc) > 0:
+            cstring = "Components"
+            if len(uc) == 1:
+                cstring = "Component"
+            warnings.append(f"WARNING: {len(uc)} {cstring} with inconsistent units")
+            next_steps.append("display_components_with_inconsistent_units()")
```

Comment:
```suggestion
            next_steps.append(self.display_components_with_inconsistent_units.__name__+"()")
```

I think we should programmatically generate these method names so that we will get an error here if the method name ever changes. With the current implementation a developer could miss this line when changing method names and wouldn't get an error or test failure.

This comment applies to every place where we are doing `next_steps.append`.